user/conditional-builds-stages-jobs.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Conditional Builds, Stages, and Jobs
+layout: en
+---
+
+## Conditional Builds, Stages, and Jobs
+
+You can filter out, and reject builds, stages, and jobs by specifying conditions in your build configuration (your `.travis.yml` file, or via [API build requests](https://docs.travis-ci.com/user/triggering-builds/)).
+
+### Conditional Builds
+
+Builds can be included or excluded by specifying a condition as follows:
+
+```yaml
+# require the branch name to be master
+if: branch = master
+
+# require the tag name to match a regular expression
+if: tag =~ ^v1
+
+# require the event type to be either `push` or `pull_request`
+if: type IN (push, pull_request)
+```
+
+Build requests that are found to be excluded will not generate a build, but will be listed on the "Requests" tab.
+
+### Conditional Stages
+
+Stages can be included or excluded by specifying a condition in the `stages` section:
+
+```yaml
+stages:
+  - name: deploy
+    # require the branch name to be master
+    if: branch = master
+
+stages:
+  - name: deploy
+    # require the tag name to match a regular expression
+    if: tag =~ ^v1
+
+stages:
+  - name: deploy
+    # require the event type to be either `push` or `pull_request`
+    if: type IN (push, pull_request)
+```
+
+At the moment, stages that are found to be excluded will be skipped silently (an improvement to this is on the roadmap).
+
+### Conditional Jobs
+
+Jobs can be included or excluded by specifying a condition on `jobs.include`:
+
+```yaml
+jobs:
+  include:
+    - # require the branch name to be master
+      if: branch = master
+      env: FOO=foo
+
+jobs:
+  include:
+    - # require the tag name to match a regular expression
+      if: tag =~ ^v1
+      env: FOO=foo
+
+jobs:
+  include:
+    - # require the event type to be either `push` or `pull_request`
+      if: type IN (push, pull_request)
+      env: FOO=foo
+```
+
+Jobs need to be listed explicitely, i.e. using `jobs.include` or `matrix.include` (alias), in order to specify conditions for them. Jobs created via [matrix expansion](/user/customizing-the-build/#Build-Matrix) currently cannot have conditions.
+
+At the moment, jobs that are found to be excluded will be skipped silently (an improvement to this is on the roadmap).
```

Comment:
Perhaps be more explicit about what sort of improvements are planned.

Reply:
👍 

Made this:

> an improvement to this is on the roadmap, giving more explicit feedback on filtering